### PR TITLE
Only reload once during the time interval

### PR DESCRIPTION
**Pull request by tomgao365**

### CODE REVIEW

**Comment by tomgao365** [comment]
I and some of my colleagues used to save multiple times in a period of time, it will open multiple windows

**Comment by tomgao365** [comment]
Although the file is saved once, it will be reload 2 times
image

**Comment by sindresorhus** [comment]
How many milliseconds are there between the `File changed` events in your screenshot? I think it would be useful to have a sensible default value for this.

Also, why is it opening two identical windows? That seems like a different bug.

File: index.js [hunk 1]
```diff
@@ -64,6 +66,14 @@ module.exports = (moduleObj, options) => {
 			console.log('File changed:', filePath);
 		}
 
+		const interval = options.interval || 0;
```
Comment (sindresorhus) [inline index.js]: The option already has a default value set.

**Comment by tomgao365** [comment]
> How many milliseconds are there between the `File changed` events in your screenshot? I think it would be useful to have a sensible default value for this.
> 
> Also, why is it opening two identical windows? That seems like a different bug.

I only saved it once

File: index.js (hunk 1 above)
Comment (tomgao365) [reply index.js]: Oh, it's my bad habit. I have modified it.

**Comment by sindresorhus** [comment]
> How many milliseconds are there between the File changed events in your screenshot? I think it would be useful to have a sensible default value for this.

?

**Comment by andrewburgess** [comment]
@sindresorhus I'm currently running into an issue that this, or something similar may be able to solve (and I suspect #3 is related).

What I'm seeing is two events fired for the same file being changed, and I believe it's because of the following actions:

1) I edit and then save a file (in VSCode)
2) Prettier formats and saves file again immediately
3) Before Electron has exited, receives two `change` events, which means `app.relaunch` is called multiple times. According to [the documentation](https://electronjs.org/docs/api/app#apprelaunchoptions), "When app.relaunch is called for multiple times, multiple instances will be started after current instance exited."

An interval, or even just a `isRelaunching` flag such as

```
if (!isRelaunching) {
  electron.app.relaunch()
  electron.app.exit(0)
}

isRelaunching = true
```

would fix it. I can submit a PR if this particular fix is not suitable

**Comment by sindresorhus** [comment]
> or even just a isRelaunching flag such as

I would prefer this solution if it works.

**Pull request closed without merging**